### PR TITLE
crystal: Fix file local variable

### PR DESCRIPTION
**Pull request by lebensterben**
- Labelled `crystal-backend` and `crystal-enable-auto-format`
  as safe local variable.
- Added local variable hooks of crystal mode:
  - `spacemacs//crystal-setup-backend`
  - `spacemacs//crystal-setup-company`
  - `spacemacs//crystal-auto-format-setup`
- Imprvoed `spacemacs//crystal-setup-backend` to setup bindings
  correctly according to backend.

See: https://github.com/syl20bnr/spacemacs/issues/14653

-------

Thank you :heart: for contributing to Spacemacs!

Before you submit this pull request, please ensure it is against the `develop` branch and that you have read the [CONTRIBUTING.org](https://github.com/syl20bnr/spacemacs/blob/develop/CONTRIBUTING.org) file. It contains instructions on how to properly follow our conventions on commit messages, documentation, change log entries and other elements.

Don't forget to update CHANGELOG.develop if you want to be mentioned in the release file!

This message should be replaced with a description of your change.

Cheers!

### CODE REVIEW

File: layers/+lang/crystal/funcs.el [hunk 1]
```diff
@@ -47,5 +47,9 @@
...
 
 (defun spacemacs//crystal-setup-backend ()
-  "Conditionally setup crystal backend."
-  (when (eq crystal-backend 'lsp) (lsp)))
+  "Conditionally setup crystal backend and bindings."
+  (if (eq crystal-backend 'lsp)
+      (lsp)
+    (spacemacs/declare-prefix-for-mode 'crystal-mode "mg" "goto")
+    (spacemacs/declare-prefix-for-mode 'crystal-mode "mt" "test")
+    (spacemacs/declare-prefix-for-mode 'crystal-mode "ma" "action")))
```
Comment (syl20bnr) [inline layers/+lang/crystal/funcs.el:55]: I know you just moved the prefix definition so this is not really related to this PR but I wonder why these prefixes are removed with the backend is `lsp`. 
Feels weird to me, `lsp` does not handle tests and it surely does not declare tests key bindings.

Comment (smile13241324) [reply layers/+lang/crystal/funcs.el:55]: This looks wrong, except of `test` these should not be bound when LSP is used. Also the prefixes should be set one time not everytime the mode is activated, this could otherwise degrade which-key's performance.

File: layers/+lang/crystal/packages.el [hunk 2]
```diff
@@ -34,7 +34,7 @@
     play-crystal))
 
 (defun crystal/post-init-company ()
-  (spacemacs//crystal-setup-company))
+  (add-hook 'crystal-mode-local-vars-hook #'spacemacs//crystal-setup-company))
```
Comment (smile13241324) [inline layers/+lang/crystal/packages.el:37]: Here we are possibly setting the company backends multiple times. Is there a side effect with company when we register a backend multiple times and what happens when we set two different backends for the same mode.

With this directory-local-variable support we seem to have to call some functions multiple times, the same may apply to prefixes and key bindings. However if we really have someone wanting to switch between lsp and non-lsp backends we will most probably not have a clean key map anymore i.e. entering a python project with anaconda, then switching to another one with lsp the prefixes from anaconda won't be removed possibly hiding prefixes from `lsp` layer.

I am not sure how to handle this @syl20bnr @lebensterben.

**Review by smile13241324 — CHANGES_REQUESTED** [review]
